src/jit/compiler.hpp

```diff
@@ -1208,10 +1208,17 @@ inline GenTree* Compiler::gtNewFieldRef(var_types typ, CORINFO_FIELD_HANDLE fldH
  *  A little helper to create an array index node.
  */
 
-inline GenTree* Compiler::gtNewIndexRef(var_types typ, GenTree* arrayOp, GenTree* indexOp)
+inline GenTree* Compiler::gtNewIndexRef(var_types typ, GenTree* arrayOp, GenTree* indexOp, bool rngChkRequired)
 {
     GenTreeIndex* gtIndx = new (this, GT_INDEX) GenTreeIndex(typ, arrayOp, indexOp, genTypeSize(typ));
 
+    assert(!(gtIndx->gtFlags & GTF_INX_RNGCHK));
+
+    if (rngChkRequired)
+    {
+        gtIndx->gtFlags |= GTF_INX_RNGCHK;
+    }
+
```

Comment:
I'd leave this as is (the flag gets set by the CTOR which is fine and normal; you should clear the flag in the importer code after the node is created, when we have no rangecheck prefix.